### DISCUSSION TITLE
How to minimise restoration of strongswan sessions after restart

### DISCUSSION THREAD

**Discussion by Jaditya-89** [strongswan/strongswan#2381] [General]
Hi, 

I am using strongswan and have ~1k sessions. When i restart the strongswan service, it takes almost 2 minutes to restore all the secure sessions. 
Is there any way to handle the configuration to minimise this restoration time as 2 minutes data traffic loss would be a huge problem.
Thank you for looking into the request

**Comment by tobiasbrunner** [comment]
Two minutes for a thousand SAs doesn't sound so bad. What's the time it takes to establish a single SA on your host? You can have a look at [this](https://docs.strongswan.org/docs/5.9/config/jobPriority.html) or [this](https://docs.strongswan.org/docs/5.9/config/lookupTuning.html) page in the docs. But note that some of these only help as responder, not as initiator. And you might want to try and determine what exactly takes the most time (key derivation, authentication, OCSP/CRL, etc.).

**Reply by Jaditya-89** [reply]
For a single SA, it takes ~1 second. But considering if the data traffic might be running during restart, then it would be huge impact on the traffic if there are scaled sessions. So for example, if the authentication part takes most time, should i increase the number of threads assigned to that process ?

**Reply by tobiasbrunner** [reply]
Depends on what exactly takes the time. If it's signature verification/creation or DH then using more threads might not really help (there can't be more operations than there are CPU cores). However, if you use e.g. RADIUS or OCSP and threads are often waiting for a response from a remote server, increasing the number of threads could help.

**Reply by Jaditya-89** [reply]
Could you please tell if there is any way to figure which part of connection establishment takes time ?

**Reply by tobiasbrunner** [reply]
You could try profiling or just look at the logs (see the [logging options](https://docs.strongswan.org/docs/5.9/config/logging.html)).
